### DISCUSSION TITLE
[iOS 18] `ipsw dyld imports` aborts with EOF

### DISCUSSION THREAD

**Discussion by lukbukkit** [blacktop/ipsw#495] [Q&A] [answered]
I'm using the `ispw dyld imports` command to search for all imports of a given framework / library. I just noticed that the command does not work with the `dyld_shared_cache_arm64e` of iOS 18.

```sh
# iOS 17.1.1 (21B91__iPhone15,4)
$ ipsw dyld imports dyld_shared_cache_arm64e Accelerate

Accelerate Imported By:
=======================

In DSC (Dylibs)
---------------
[...]

In FileSystem DMG (Apps)
------------------------
[...]
```

```sh
# iOS 18 (22A5282m__iPhone16,2)
$ ipsw dyld imports dyld_shared_cache_arm64e Accelerate

Accelerate Imported By:
=======================
Usage:
  ipsw dyld imports <DSC> <DYLIB> [flags]

Aliases:
  imports, imp

Flags:
  -h, --help          help for imports
  -i, --ipsw string   Path to IPSW to scan for MachO files that import dylib

Global Flags:
      --color           colorize output
      --config string   config file (default is $HOME/.config/ipsw/config.yaml)
      --no-color        disable colorize output
  -V, --verbose         verbose output

   ⨯ failed to get dylibs that import /System/Library/Frameworks/Accelerate.framework/Accelerate: EOF
```

Am I doing something wrong, or is this a novel issue with the dyld cache of iOS 18? Thanks for building such great iOS tooling :)

Best
Lukas

**Comment by blacktop** [comment, marked as answer]
looks like a decent amount of changes in iOS 18.  made a feat branch [feat/prebuilt_iOS18](https://github.com/blacktop/ipsw/tree/feat/prebuilt_iOS18)

**Comment by blacktop** [comment]
Just an update, I've merged the branch as a FIX PR for now, so that it properly errors for people and fixes the `ipsw dsc imports` fall-back to just scanning DSC dylib LCs

**Comment by blacktop** [comment]
@lukbukkit this should be fixed in latest now.

**Reply by lukbukkit** [reply]
Thanks a lot :)
